### PR TITLE
[DSL][go-gen] Remove #enumerableByThis

### PR DESCRIPTION
**Pull request by c-harding**
This removes the #enumerableByThis metadata, because this would be business logic to implement. To continue development in removing this, simply remove the field `filterEnumeration` from `CreatedByAttribute`, and following the errors.

### CODE REVIEW

File: src/main/scala/temple/builder/ServerBuilder.scala [hunk 1]
```diff
@@ -31,11 +31,11 @@ object ServerBuilder {
     }
 
     val createdBy: Option[CreatedByAttribute] = serviceBlock.lookupMetadata[ServiceEnumerable].map {
-      case ServiceEnumerable(byThis) =>
+      case ServiceEnumerable =>
         CreatedByAttribute(
           languageConfig.createdByInputName,
           languageConfig.createdByName,
-          filterEnumeration = byThis,
+          filterEnumeration = false,
```
Comment (willjvsmith) [inline src/main/scala/temple/builder/ServerBuilder.scala]: I'll be fixing this in my PR

**Review by willjvsmith — APPROVED** [review]

**Review by jaylees14 — APPROVED** [review]

File: src/main/scala/temple/builder/ServerBuilder.scala (hunk 1 above)
Comment (willjvsmith) [reply src/main/scala/temple/builder/ServerBuilder.scala]: nvm fixed it in this one

**Review by willjvsmith — APPROVED** [review]
Great work

**Pull request merged**